protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+
+2. **Staged commit** <a name="staged-commit"></a>: A proposed commit that is written to disk at
+   `_delta_log/_staged_commits/<v>.<uuid>.json`.
+    - Here, the `uuid` is a random UUID that is generated for each commit and `v` is the version
+      which is being committed, zero-padded to 20 digits.
+    - The catalog tracks only the location, not the content, of a staged commit.
+    - The mere existence of one of these files does not mean that the file is a _ratified_ commit.
+      It might correspond to a failed or in-progress commit attempt.
+    - The catalog is the source of truth around which staged commits are "ratified".
+
+3. **Inline commit** <a name="inline-commit"></a>: A proposed commit that is not written to disk but
+   rather has its content sent to the catalog for the catalog to track directly.
+
+4. **Ratified commit**: A proposed commit that a catalog has determined has won the commit at the
+   desired version of the table.
+    - The catalog must track ratified commits until they are published to the `_delta_log` directory.
+    - A ratified commit may or may not yet be published.
+    - A ratified commit may or may not even be _tracked_ by the catalog at all - the catalog may
+      have just atomically published it to the filesystem directly, relying on PUT-if-absent
+      primitives to facilitate the ratification and publication all in one step.
+
+5. **Published commit** <a name="published-commit"></a>: A ratified commit that has been copied into
+   the `_delta_log` as a normal Delta file, i.e. `_delta_log/<v>.json`.
+    - Here, the `v` is the version which is being committed, zero-padded to 20 digits.
+    - The existence of a `<v>.json` file proves that the corresponding version `v` is ratified,
+      regardless of whether the table is catalog-owned for filesystem-based. The catalog is allowed
+      to return information about published commits, but Delta clients can also use filesystem
+      listing operations to directly discover them.
+    - Published commits do not need to be tracked by the catalog.
+
+## Terminology: Delta Client
+
+This is the component that implements support for reading and writing Delta tables, and implements
+the logic required by the `catalogOwned` table feature. Among other things, it
+- triggers the filesystem listing, if needed, to discover published commits
+- generates the commit content (the set of [actions](#actions))
+- works together with the query engine to trigger the commit process and invoke the client-side
+  catalog component with the commit content
+
+## Terminology: Catalogs
+
+1. **Catalog**: A catalog is a entity which manages a Delta table, including its creation, writes,
+   reads, and eventual deletion.
+    - It could be backed by a database, a filesystem, or any other persistence mechanism.
+    - Each catalog has its own spec around how Delta clients should contact them, and how they
```

Comment:
"contact them" implies the presence of a server. How about "interact with them"?